streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/testing/TestRunSourceSpout.java

```diff
@@ -90,12 +85,13 @@ public void nextTuple() {
         int emitCount = 0;
 
         // loop output stream and emit at most one record per output stream
-        for (Map.Entry<String, Queue<Map<String, Object>>> entry : testRecordsQueueMap.entrySet()) {
+        for (Map.Entry<String, TestRecordsInformation> entry : testRecordsInformationPerOutputStream.entrySet()) {
             String outputStream = entry.getKey();
-            Queue<Map<String, Object>> queue = entry.getValue();
+            TestRecordsInformation info = entry.getValue();
 
-            Map<String, Object> record = queue.poll();
-            if (record != null) {
+            Optional<Map<String, Object>> recordOptional = info.nextRecord();
```

Comment:
will it busy loop? Any reason for not having a `sleep(duration)` after the test case data is emitted?

Reply:
Because we are handling multiple output streams and sleep duration for each could be different. We don't loop in busy whenever there're no record to emit (whole iterations are completed or it's in sleep phase).

Reply:
isn't the sleep per source ?

Reply:
Ah yes you're right. I was a bit confused.
But separate records per output stream still holds true, so `sleep(duration)` doesn't also ensure exact sleep. This may be OK.

Reply:
Addressed some logic to respect the duration.